controller/volume_controller.go

```diff
@@ -3170,6 +3173,26 @@ func (c *VolumeController) checkAndFinishVolumeRestore(v *longhorn.Volume, e *lo
 		return nil
 	}
 
+	// Make sure the backup volume is up-to-date and the latest backup is restored
+	_, bvName, _, err := backupstore.DecodeBackupURL(v.Spec.FromBackup)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get backup name from volume %s backup URL %v", v.Name, v.Spec.FromBackup)
+	}
+	bv, err := c.ds.GetBackupVolume(bvName)
+	if err != nil {
+		return err
+	}
+	if !bv.Status.LastSyncedAt.IsZero() &&
+		bv.Spec.SyncRequestedAt.After(bv.Status.LastSyncedAt.Time) {
+		log.Infof("Restore/DR volume needs to wait for backup volume %s update", bvName)
+		return nil
```

Comment:
Should we requeue the volume?

Reply:
Here the volume controller should wait for the backup volume controller to sync up this `bv`.